### DISCUSSION TITLE
Station classification

### DISCUSSION THREAD

**Discussion by freddepettersson** [tellurianinteractive/Tellurian.Trains.ModulesRegistryApp#76] [Ideas]
Hi
Add different station typs.
( sorry need to use swedish)

- lager station = Driftplats
- small hop on/ off = Hållplats
- cargo place = Linjeplats
- magasin = Bangårdsområde

Of course it should be translated into all language.
The more specific stationtype will give meetingplaner a better indication of station size

/ Fredde

**Comment by fjallemark** [comment]
I have the same thoughts. In Sweden, the classification is defined by the Swedish Transport Agency (Transportstyrelsen). I present the classification in Swedish and a proposed translation to English based on the *Rail Lexicon*.

They make a distinction between a **operated place** (*trafikplats*) and **the line** (*linjen*).  
An **operated place** (*trafikplats*) is a common term for *operated site*, *operated site part*, *line site*, *stop* and *halt*. The subterms are:
- **Operated site** (*driftplats*) is an area delimited by the line which can be monitored by train dispatchers in more detail than
what is required for the line.
- **Operated site part** (*driftplatsdel*) is a delimited part of an *operated site*, usually occuring only where there is a technical need to divide an *operating site* into several named parts.
- **Line site** (*linjeplats*) is a place on the line with at least one switch or variable bridge in the main track. A line site can also be intended for boarding and alighting. There may be *line sites* without main track, where the main track turns into siding at the line location boundary and where all switches are within the *line site* limit.
- **Stop** (*hållplats*) is a place on the line, intended for boarding and alighting, but without any switch or variable bridge in the main track.
- **Halt** (*hållställe*) is a place with a special name within an operating place intended
for boarding and alighting.

In the Module Registry, a station *is a place where train times will be recorded*. This means that a module must be defined as a station if it fulfils any of the definitions above. However, these definitions does not say anything about the *purpose* of the station. 

Here a suggestion for capabilities of module stations:
- **Has main line switches** is a station module that has switches to, from or between any scheduled track, which indicates that the station must be monitored. This could be a check-box to tick.
- **Has passenger exchange** means that passenger trains can stop here. If any track has a *platform length* this is assumed to be the case. The platform length defines the maximum passenger train length (minus the length of the loco).
- **Has shunting capacity** means that freight trains can be shunted and re-arranged at the station.  This could be a check-box to tick.
- **Is Terminus** means that the line ends at the station module. There is already a check-box for this.

From the capabilities above, you can deduct the classification for the *purpose*, i.e. the *usage* of an operation site:
- **Line site** is one that *has main line switches* and optionally *is terminus*,
- **Stop** is one that *has passenger exchange* and optionally *is terminus*,
- **Station** is a *line site* that also *has passenger exchange* and optionally *is terminus*,
- **Yard** is *line site* that also *has shunting capacity*,
- **Main station** is a *station* that also *has shunting capacity*,
- **Shadow station** is a *main station* that  also *is terminus*.

**Comment by JanneFrelin** [comment]
That what a lot of hassle just to take part in a discussion...

What reference did you use at Trafikverket.

**Reply by fjallemark** [reply]
Welcome Janne! Thank you for all the hassle to be a GitHub user. 😊

The reference comes from the **Swedish Transport Agency** (*Transportstyrelsen*), not the **Swedish Transport Administration** (*Trafikverket*).  The terminology is part of the **Rail Traffic Regulation Act** (*Järnvägstrafiksförordningen JTF*) and its first chapter [**Terminology** (*Termer*)](https://www.transportstyrelsen.se/globalassets/global/publikationer/jarnvag/jtf/pj09002_jtf_1_termer.pdf).
